### PR TITLE
Start new chat with favorites from recent screen.

### PR DESCRIPTION
**Pull request by roberthein**
img_4228
img_4229

### CODE REVIEW

File: Toshi/Controllers/Favorites/FavoritesController.swift [hunk 1]
```diff
@@ -145,9 +147,13 @@ open class FavoritesController: SweetTableController {
         } else {
             tableView.tableHeaderView = searchController.searchBar
         }
-
-        navigationItem.rightBarButtonItem = addButton
-
+        
+        if isPresentedModally {
+            navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(didPressCancel(_:)))
```
Comment (yuliaveres) [inline Toshi/Controllers/Favorites/FavoritesController.swift]: maybe lets create this one in the same way as we do with `addButton` - lazy

File: Toshi/Controllers/Favorites/FavoritesController.swift [hunk 2]
```diff
@@ -84,6 +84,10 @@ open class FavoritesController: SweetTableController {
 
         return controller
     }()
+    
+    private var isPresentedModally: Bool {
```
Comment (yuliaveres) [inline Toshi/Controllers/Favorites/FavoritesController.swift]: is that second case we need it ? maybe lets have some convenience ? extension or smth, and use it on any view controller ?

File: Toshi/Controllers/Favorites/FavoritesController.swift [hunk 3]
```diff
@@ -165,9 +171,15 @@ open class FavoritesController: SweetTableController {
             }
         }
     }
-
+    
+    @objc func didPressCancel(_ barButtonItem: UIBarButtonItem) {
```
Comment (yuliaveres) [inline Toshi/Controllers/Favorites/FavoritesController.swift]: private ? 😉

File: Toshi/Controllers/Favorites/FavoritesController.swift [hunk 4]
```diff
@@ -452,14 +464,27 @@ extension FavoritesController: UITableViewDelegate {
...
-
-            UserDefaults.standard.setValue(contact.address, forKey: FavoritesNavigationController.selectedContactKey)
+            
+            if isPresentedModally {
+                self.searchController.isActive = false
+                ChatsInteractor.getOrCreateThread(for: contact.address)
+                
+                DispatchQueue.main.async {
+                    guard let delegate = UIApplication.shared.delegate as? AppDelegate, let window = delegate.window else { return }
+                    (window.rootViewController as? TabBarController)?.displayMessage(forAddress: contact.address)
```
Comment (yuliaveres) [inline Toshi/Controllers/Favorites/FavoritesController.swift]: you can do `Navigator.tabbarController.displayMessage(forAddress: contact.address)` 😉

File: Toshi/Controllers/Messaging/ChatsController.swift [hunk 5]
```diff
@@ -102,6 +102,15 @@ open class ChatsController: SweetTableController {
 
         preferLargeTitleIfPossible(true)
         tabBarController?.tabBar.isHidden = false
+        
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .compose, target: self, action: #selector(didPressCompose(_:)))
+
```
Comment (yuliaveres) [inline Toshi/Controllers/Messaging/ChatsController.swift]: **notsureif**

File: Toshi/Controllers/Messaging/ChatsController.swift [hunk 6]
```diff
@@ -102,6 +102,15 @@ open class ChatsController: SweetTableController {
 
         preferLargeTitleIfPossible(true)
         tabBarController?.tabBar.isHidden = false
+        
+        navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .compose, target: self, action: #selector(didPressCompose(_:)))
+        
+        
+    }
+    
+    @objc func didPressCompose(_ barButtonItem: UIBarButtonItem) {
```
Comment (yuliaveres) [inline Toshi/Controllers/Messaging/ChatsController.swift]: private ? 😉

**Comment by yuliaveres** [comment]
Also, please, double check `FavoritesController 164` `if let address = UserDefaults.standard.string(forKey: FavoritesNavigationController.selectedContactKey) {..` this should happen only on `Favorites` tab

File: Toshi/Controllers/Favorites/FavoritesController.swift (hunk 2 above)
Comment (roberthein) [reply Toshi/Controllers/Favorites/FavoritesController.swift]: This is such a small thing, we don't need an extension for everything right? Also the implementation could vary..

File: Toshi/Controllers/Favorites/FavoritesController.swift (hunk 4 above)
Comment (roberthein) [reply Toshi/Controllers/Favorites/FavoritesController.swift]: 💥

File: Toshi/Controllers/Messaging/ChatsController.swift (hunk 5 above)
Comment (roberthein) [reply Toshi/Controllers/Messaging/ChatsController.swift]: What are you not sure about @yuliaveres?

Comment (yuliaveres) [reply Toshi/Controllers/Messaging/ChatsController.swift]: @roberthein  two  empty lines 😕

Comment (roberthein) [reply Toshi/Controllers/Messaging/ChatsController.swift]: 😭

File: Toshi/Controllers/Favorites/FavoritesController.swift (hunk 1 above)
Comment (roberthein) [reply Toshi/Controllers/Favorites/FavoritesController.swift]: done!

File: Toshi/Controllers/Messaging/ChatsController.swift (hunk 6 above)
Comment (roberthein) [reply Toshi/Controllers/Messaging/ChatsController.swift]: yes!

**Pull request merged**